src/components/form/Form.tsx

```diff
@@ -1,161 +1,94 @@
-      {/*for now there are only two fields for testing purposes, later the rest will be added*/}
-      <StyledForm onSubmit={handleSubmit}>
-        <div>
-          <LabelField>
-            podaj nazwę roślinki
-            <InputField
-              height={"50px"}
-              placeholder={"nazywam się..."}
-              type={"text"}
-              name={"plantName"}
-              value={singlePlant.plantName}
-              handleChange={handleChange}
-              onKeyDown={handleKeyDown}
-            />
-          </LabelField>
-        </div>
-        <div>
-          <LabelField>
-            dodaj jej zdjęcie
-            <InputField
-              height={"190px"}
-              placeholder={
-                "zalecane wymiary to min. 250px na 200px, format png  lub jpg"
-              }
-              type={"file"}
-              accept="image/png, image/jpeg"
-              name={"plantImage"}
-              handleChange={handleChange}
-              value={singlePlant.plantImage}
-            />
-          </LabelField>
-        </div>
-
+      <StyledForm onSubmit={handleSubmit(onSubmit)}>
+        <LabelField>
+          podaj nazwę roślinki
+          <InputField
+            height={"50px"}
+            placeholder={"nazywam się..."}
+            type={"text"}
+            {...register("plantName")}
+          />
+        </LabelField>
+
+        <LabelField>
+          jak chcesz ją podlewać
+          <InputField
+            height={"50px"}
+            placeholder={"wpisz jak bardzo lubię wodę..."}
+            type={"text"}
+            {...register("watering")}
+          />
+        </LabelField>
+        <LabelField>
+          czy lubi zraszanie
+          <InputField
+            height={"50px"}
+            placeholder={
+              "niektóre z nas to uwielbiają, a inne \n" +
+              "nie mogą znieść, a ja..."
+            }
+            type={"text"}
+            {...register("misting")}
+          />
+        </LabelField>
         <FormButton
           type={"submit"}
           color={Color.cream}
           backgroundColor={Color.mediumGreen}
-          handleClick={submitFormData}
         >
           zapisz
         </FormButton>
-      </StyledForm>
-
-      <h4>from local storage</h4>
-      <p style={{ color: Color.cream }}>
-        To check if data is correctly received. For now it's not really working.
-        The context will be created later on and initialFormData mapped in
-        correct places
-      </p>
 
-      {initialFormData.plants.map((item, index) => {
-        return (
-          <NavItem
-            backgroundColor={Color.lightGreen}
-            color={Color.mediumGreen}
-            key={index}
-            linkTo={`/${item.plantName}`}
-          >
-            {item.plantName}
-          </NavItem>
-        );
-      })}
+        <h4>plants from local storage</h4>
+
+        {plantsArray.map((item, index) => {
+          return (
+            <div key={index}>
```

Comment:
iteration index as key is generally discouraged - may lead to unexpected list behavior. Do you have any other identifier available here?

Reply:
oh right! I used plantID then :)